UPGRADE-3.1.md

```diff
@@ -64,6 +64,9 @@ FrameworkBundle
     - `"form.type.submit"`
     - `"form.type.reset"`
 
+ * The `Controller::getUser()` method has been deprecated and will be removed in
+   Symfony 4.0; typehint the security user object in the action instead.
```

Comment:
I think this should make clear that you must use the `UserInterface` from the Security component for the type hint. Otherwise, people may think they can type hint using their concrete implementations.